### PR TITLE
Add 'drum-circle' example from ology's repo, docs updates

### PR DESCRIPTION
**Pull request by jbarrett**

### CODE REVIEW

**Comment by jbarrett** [comment]
@ology Hope I didn't butcher your code too badly 😬

**Review by ology — APPROVED** [review]
- This works as-is on windows 11.

- To run it on my Linux machine, I have to `timidity -iAD` in another terminal first.

- To run it on a Mac, copy an sf2 soundfont file into the `~/Library/Audio/Sounds/Banks/` directory, Then a DAW app, like Logic Pro, has to be open and a track created with the Apple AU Instrument "DLSMusicDevice". And in the settings for that instrument, the "Sound Bank" must be set to the soundfont that was copied, earlier. WHEW :\

**Comment by ology** [comment]
So... `$opts{device}->open_virtual_port('foo') if $^O eq 'darwin';`

File: examples/drum-circle [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+    sub reset_score {
+        my ( $self, $drummer ) = @_;
+        # sorry
+        $drummer->score->{ Score } = [
+            grep { $_->[0] !~ /^note/ }
+            @{ $drummer->score->{ Score } }
+        ];
+        ${ $drummer->score->{ Time } } = 0;
+        delete $self->{common}->{seen};
```
Comment (ology) [inline examples/drum-circle:155]: This should be `$self->{common}->{seen} = {};`

File: examples/drum-circle [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+        }
+        bless \%opts, $class;
+    }
+
+    sub device { shift->{ device } }
+
+    # This manipulates internals of MIDI::Score objects and
+    # hashes used by drum-circle - doing this isn't a good
+    # idea - skip to `sub play` to see the interesting piece
+    # of this example.
```
Comment (jbarrett) [inline examples/drum-circle:166]: Admonishment!

**Comment by jbarrett** [comment]
Many thanks @ology! I also added some docs outlining your DLSMusicDevice approach, alongside fluidsynth (don't know if you want to check that particular one out - like timidity with full Mac support as far as I can tell. Works well on Linux)

**Comment by jbarrett** [comment]
Many thanks @ology! Going to kick this into the neighbour's yard, see what they make of it.

File: README.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ $device->open_virtual_port( 'foo' );
 # MIDI device on your system, such as a loopback device, or virtual or
 # hardware synth. Your device must be connected to some sort of synth to
 # make noise.
-$device->open_port_by_name( qr/wavetable|loopmidi|timidity|dls/i );
+$device->open_port_by_name( qr/wavetable|loopmidi|timidity/i );
```
Comment (ology) [inline README.md:22]: Doesn't this need an `|fluid` too?

**Review by ology — COMMENTED** [review]
👍🏻

**Pull request merged**